### PR TITLE
Fix warnings, improve performance

### LINKED ISSUE #1639: Can not get local trailers to play in webinterface
**Describe the bug**
Local trailers are found in the function "RefreshLocalTrailers", but are not playable in the webinterface.  LocalTrailerIds contains an GUID but LocalTrailerCount is 0.
As an result a youtube trailer is played.

**To Reproduce**
Create a trailers folder and put an File into this folder e.g.:
/Movie (2001)/Movie (2001).mkv
/Movie (2001)/trailers/Movie (2001).mkv
Rescan the Lib
Try to play the trailer in the webinterface

**System (please complete the following information):**
 - OS:  Windows
 - Browser: Chrome
 - Jellyfin Version: Commit 6766e04
 - Reverse proxy:  no

### PR DESCRIPTION
**Pull request by Bond-009**
`QueryResult.Items` is now a `IReadOnlyList` so we don't need to
allocate a new `Array` when we have a `List` (and `Items` shouldn't need to
be mutable anyway)

This also fixes #1639

### CODE REVIEW

**Comment by cvium** [comment]
`Manager/ProviderUtils.cs(200,43): error CS8107: Feature 'default literal' is not available in C# 7.0. Please use language version 7.1 or greater. [/home/vsts/work/1/s/MediaBrowser.Providers/MediaBrowser.Providers.csproj]`

**Comment by dkanada** [comment]
I would rather rebase than leave a needless merge commit in the history.

File: Emby.Server.Implementations/Dto/DtoService.cs [hunk 1]
```diff
@@ -1074,17 +1067,25 @@ private void AttachBasicFields(BaseItemDto dto, BaseItem item, BaseItem owner, D
...
             if (options.ContainsField(ItemFields.LocalTrailerCount))
             {
+                int trailerCount = 0;
                 if (allExtras == null)
                 {
                     allExtras = item.GetExtras().ToArray();
                 }
 
-                dto.LocalTrailerCount = allExtras.Count(i => i.ExtraType.HasValue && i.ExtraType.Value == ExtraType.Trailer) + item.GetTrailers().Count();
+
```
Comment (cvium) [inline Emby.Server.Implementations/Dto/DtoService.cs]: Needless newline
```suggestion
```

File: MediaBrowser.Api/UserLibrary/UserLibraryService.cs [hunk 2]
```diff
@@ -367,12 +368,21 @@ public object Get(GetLocalTrailers request)
             var dtoOptions = GetDtoOptions(_authContext, request);
 
             var dtosExtras = item.GetExtras(new[] { ExtraType.Trailer })
-                .Select(i => _dtoService.GetBaseItemDto(i, dtoOptions, user, item));
```
Comment (cvium) [inline MediaBrowser.Api/UserLibrary/UserLibraryService.cs]:
```suggestion
```

**Pull request merged**